src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/LibraryImportGenerator.cs

```diff
@@ -313,6 +313,22 @@ private static SyntaxTokenList StripTriviaFromModifiers(SyntaxTokenList tokenLis
                 return (PrintForwarderStub(pinvokeStub.StubMethodSyntaxTemplate, explicitForwarding: true, pinvokeStub, diagnostics), pinvokeStub.Diagnostics.Array.AddRange(diagnostics.Diagnostics));
             }
 
+            bool supportsTargetFramework = true;
+            if (pinvokeStub.LibraryImportData.SetLastError && !options.GenerateForwarders)
+            {
+                supportsTargetFramework = pinvokeStub.GeneratorFactoryKey.Key.TargetFramework == TargetFramework.Net
+                    && pinvokeStub.GeneratorFactoryKey.Key.Version.Major >= 6;
+            }
```

Comment:
Should we just add the conditions for the if block to the `supportsTargetFramework` assignment?

```suggestion
            bool supportsTargetFramework = pinvokeStub.LibraryImportData.SetLastError
                && !options.GenerateForwarders
                && pinvokeStub.GeneratorFactoryKey.Key.TargetFramework == TargetFramework.Net
                && pinvokeStub.GeneratorFactoryKey.Key.Version.Major >= 6;
```